### PR TITLE
Structured help page formatting

### PR DESCRIPTION
**Pull request by stephenfin**
The current help page is a little hard to parse as there's no hierarchy of information. Attempt to improve this by making use of headers, in the absence of other structure elements (e.g. horizontal lines) in Slack's Markdown, to produce a manpage-like format.

Before:

---

`rosa` `lookup` `version`

_Find openshift version(s) with provided prefix that is supported in ROSA._

> *Example:* _rosa lookup 4.15_

---

After:

---

# rosa lookup
## Usage
```
rosa lookup version
```
## Description
Find openshift version(s) with provided prefix that is supported in ROSA.
## Example
```
rosa lookup 4.15
```

---

### CODE REVIEW

File: pkg/slack/events/messages/message_handler.go [hunk 1]
```diff
@@ -84,26 +84,51 @@ func postResponse(client *slack.Client, event *slackevents.MessageEvent, respons
...
 
 func help(client *slack.Client, event *slackevents.MessageEvent, botCommands []parser.BotCommand) {
-	helpMessage := " "
-	helpMessage += "help" + " - " + fmt.Sprintf("_%s_", "help") + "\n"
+	helpMessage := ""
 	for _, command := range botCommands {
 		tokens := command.Tokenize()
+
+		// # <command>
+		helpMessage += "# "
```
Comment (bradmwilliams) [inline pkg/slack/events/messages/message_handler.go:92]: Unfortunately, Slack's `mrkdwn` language does not fully support everything that traditional Markdown does, like Headings:
https://api.slack.com/reference/surfaces/formatting#basics

Comment (bradmwilliams) [reply pkg/slack/events/messages/message_handler.go:92]: Here is how your current changes render:
image

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: pkg/slack/events/messages/message_handler.go (hunk 1 above)
Comment (stephenfin) [reply pkg/slack/events/messages/message_handler.go:92]: Damn, I really should have sent some messages to myself/RTM rather than relying on third-party blogs. Let me think on how to rework this further...

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: pkg/slack/events/messages/message_handler.go [hunk 2]
```diff
@@ -84,26 +84,51 @@ func postResponse(client *slack.Client, event *slackevents.MessageEvent, respons
...
 
 func help(client *slack.Client, event *slackevents.MessageEvent, botCommands []parser.BotCommand) {
-	helpMessage := " "
-	helpMessage += "help" + " - " + fmt.Sprintf("_%s_", "help") + "\n"
+	helpMessage := ""
 	for _, command := range botCommands {
 		tokens := command.Tokenize()
+
+		// # <command>
+		helpMessage += ":black_circle: *"
```
[the remark this answers is not in the document]
Comment (stephenfin) [reply pkg/slack/events/messages/message_handler.go:92]: wfm. I'll change that and update the test

[the remark this answers is not in the document]
Comment (stephenfin) [reply pkg/slack/events/messages/message_handler.go]: Done. I forgot the test failure was due to the quota issue.
